portfolio/src/main/webapp/slideshow_script.js

```diff
@@ -0,0 +1,55 @@
+//sliding
+var slideIndex = 1;
+var timeout;
+showSlide(slideIndex); //initial state
+
+function showSlide(index){
+    var slides = document.getElementsByClassName("slide");
+    var noSlides = slides.length
+    console.log(noSlides);
+    if(index>noSlides || index<1 ){
+        index=(index%noSlides+noSlides)%noSlides+1;
+    }
+    for(var i=0; i<noSlides; i++){
+        slides[i].style.display="none";
+    }
+    slides[index-1].style.display="block";
+    console.log("show slide: " + index);
+}
+
+function plusSlides(n){
+    slideIndex=slideIndex+n;
+    showSlide(slideIndex);
+}
+
+// fading buttons, captions and slidenumbers
+var fadingObjects = document.querySelectorAll('.btn-prev, .btn-next, .caption, .slidenumber')
+setFadingObjectsEventListener();
+
+function showButtons(){
+    clearTimeout(timeout);
+    console.log("show buttons");
+    for (var i = 0; i < fadingObjects.length; i++) {
+        fadingObjects[i].style.display="block";
+    }
+    timeout = setTimeout(hideButtons, 3000);
+}
+
+function fadeObject(object){
+    for(var i=10; i>=1; i--){
+        object.style.opacity=i/10;
```

Comment:
Is this fade noticeable? If the machine is quite fast at doing this you will probably only see fully opaque and then transparent? Shouldn't we do something time-based?

Reply:
No, this function wasn't even called, it didn't have any effect. I saw that this fading should have been done using jQuery, which seemed to be out of the scope, so in the end I decided to let the objects disappear at once instead. I deleted this function.